### PR TITLE
[Merged by Bors] - chore(Rat/Cardinal): reduce imports and create new file

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---
I'm going to use Continued Fraction for denumerability of Q. We don't need continued fractions to prove the cardinality of Q and we don't need cardinals to prove denumerability of Q. So I seperated the files, and made a new file for the lemma about the cardinality of Q.

I also golfed the proof of denumerability, it's slightly shorter if I use the proof of encodability.

The proof of `Cardinal.mkRat` is just `Cardinal.mk_eq_aleph0`, however the set of imports required to make this proof work is not obvious, so I think it's still worth keeping the lemma.

I wasn't sure what to do with the copyright header in the new file. I just copied the one from the `denumerable` file and added Floris Van Doorn as an author as he wrote the cardinality lemma.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Rat/Denumerable.lean [hunk 1]
```diff
@@ -3,13 +3,15 @@ Copyright (c) 2019 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes
 -/
-import Mathlib.Algebra.Order.Ring.Rat
-import Mathlib.SetTheory.Cardinal.Basic
+import Mathlib.Algebra.CharZero.Infinite
+import Mathlib.Algebra.Ring.Rat
+import Mathlib.Data.Rat.Encodable
+import Mathlib.Logic.Denumerable
```
Comment (vihdzp) [inline Mathlib/Data/Rat/Denumerable.lean:9]: I imagine these imports should be enough? (I haven't checked)
```suggestion
import Mathlib.Data.Rat.Encodable
import Mathlib.Logic.Denumerable
```

File: Mathlib/Data/Rat/Cardinal.lean [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Floris Van Doorn
+-/
+import Mathlib.Algebra.CharZero.Infinite
+import Mathlib.Algebra.Ring.Rat
+import Mathlib.Data.Rat.Encodable
+import Mathlib.SetTheory.Cardinal.Basic
```
Comment (vihdzp) [inline Mathlib/Data/Rat/Cardinal.lean:9]: Likewise:
```suggestion
import Mathlib.Data.Rat.Encodable
import Mathlib.SetTheory.Cardinal.Basic
```

**Review by vihdzp — COMMENTED** [review]
Mostly LGTM, though you seem to be running with some issues with imports. Something you can do is replace every instance of `import Mathlib.Data.Rat.Denumerable` by `import Mathlib.Data.Rat.Cardinal`, which should get Mathlib to build, then use `lake exe shake` to remove the now unneeded imports.

File: Mathlib/Data/Rat/Cardinal.lean (hunk 2 above)
Comment (ChrisHughes24) [reply Mathlib/Data/Rat/Cardinal.lean:9]: That's not enough imports. I did the `lake exe shake` command as you suggested

File: Mathlib/Data/Rat/Denumerable.lean [hunk 3]
```diff
@@ -3,13 +3,15 @@ Copyright (c) 2019 Chris Hughes. All rights reserved.
...
 
 /-!
 # Denumerability of ℚ
 
-This file proves that ℚ is infinite, denumerable, and deduces that it has cardinality `omega`.
+This file proves that ℚ is denumerable.
 -/
```
Comment (kim-em) [inline Mathlib/Data/Rat/Denumerable.lean:15]: Could you add a reference to the new `Rat/Cardinal.lean` file here?

**Comment by kim-em** [comment]
Otherwise,

bors d+

**Comment by ChrisHughes24** [comment]
Bors r+

**Pull request closed without merging**